src/parse_fc.c

```diff
@@ -268,18 +306,51 @@ struct policy_node *parse_fc_file(const char *filename, const struct string_list
 			continue;
 		}
 
-		struct fc_entry *entry = parse_fc_line(line);
+		if (is_within_ifdef) {
+			conditional = malloc(sizeof(struct conditional_data));
+			conditional->flavor = CONDITION_IFDEF;
+			conditional->condition = strdup(ifdef_condition);
+		} else if (is_within_ifndef) {
+			conditional = malloc(sizeof(struct conditional_data));
+			conditional->flavor = CONDITION_IFNDEF;
+			conditional->condition = strdup(ifndef_condition);
```

Comment:
These strings will get leaked if there is a parse error in the line (the fc_entry node will never take ownership of them.

You either need to check for (and free) them in the cleanup tag in parse_fc_line(), or sidestep all the memory management issues here by just passing the flavor and string to parse_fc_line directly.